pyproject.toml

```diff
@@ -45,31 +45,35 @@ testpaths = "cf_units"
 write_to = "cf_units/_version.py"
 local_scheme = "dirty-tag"
 
-[tool.black]
+[tool.ruff]
 line-length = 79
-target-version = ["py39", "py310", "py311"]
-include = '\.pyi?$'
-exclude = '''
-(
-  /(
-      \.eggs         # exclude a few common directories in the
-    | \.git          # root of the project
-    | \.hg
-    | \.mypy_cache
-    | \.tox
-    | \.venv
-    | _build
-    | buck-out
-    | build
-    | dist
-  )/
-  | _version.py
-)
-'''
+target-version = "py38"
+select = [
+    # pyflakes
+    "F",
+    # pycodestyle
+    "E",
+    "W",
+    # flake8-bugbear
+    "B",
+    # flake8-comprehensions
+    "C4",
+    # isort
+    "I",
+    # pyupgrade
+    "UP",
+]
+ignore = ["B904", "F403", "F405"]
+
+[tool.ruff.isort]
+known-first-party = ["cf_units"]
+
+[tool.ruff.mccabe]
+max-complexity = 22
+
+[tool.ruff.pydocstyle]
+convention = "numpy"
 
-[tool.isort]
-known_first_party = "cf_units"
-line_length = 79
-profile = "black"
-skip_gitignore = "True"
-verbose = "False"
+[tool.ruff.per-file-ignores]
+# Allow the longer comment lines that the generated code produces.
+"cf_units/_udunits2_parser/parser/*.py" = ["E501"]
```

Comment:
The alternative way of blanket ignoring everything with ruff-formatter is to put this exclude in ``pre-commit``

Reply:
To be clear, the changes in these files came from the ruff linter (which has some auto-fixes) rather than ruff formatter.

I have no objection to skipping linting/formatting for these files but, since they were previously flake8-linted and black-formatted I feel like that decision has already been made for the current generation.  If whoever does the next re-generation wants to turn off the linting and their reviewer agrees then that's a decision for then.